sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -0,0 +1,907 @@
+
+            DistributionSummary requestChargeMeter = DistributionSummary
+                .builder(nameOf("op.RUs"))
+                .baseUnit("RU (request unit)")
+                .description("Operation RU charge")
+                .maximumExpectedValue(10_000_000d)
+                .publishPercentiles(0.95, 0.99)
+                .publishPercentileHistogram(true)
+                .tags(operationTags)
+                .register(compositeRegistry);
+            requestChargeMeter.record(Math.max(requestCharge, 10_000_000d));
+
+            Timer latencyMeter = Timer
+                .builder(nameOf("op.latency"))
+                .description("Operation latency")
+                .maximumExpectedValue(Duration.ofSeconds(300))
+                .publishPercentiles(0.95, 0.99)
+                .publishPercentileHistogram(true)
+                .tags(operationTags)
+                .register(compositeRegistry);
+            latencyMeter.record(latency);
+
+            this.recordItemCounts(maxItemCount, actualItemCount);
+
+            List<ClientSideRequestStatistics> clientSideRequestStatistics =
+                diagnosticsAccessor.getClientSideRequestStatistics(diagnostics);
+
+            if (clientSideRequestStatistics != null) {
+                for (ClientSideRequestStatistics requestStatistics : clientSideRequestStatistics) {
+
+                    recordStoreResponseStatistics(requestStatistics.getResponseStatisticsList());
+                    recordStoreResponseStatistics(requestStatistics.getSupplementalResponseStatisticsList());
+                    recordGatewayStatistics(requestStatistics.getDuration(), requestStatistics.getGatewayStatistics());
+                    recordAddressResolutionStatistics(requestStatistics.getAddressResolutionStatistics());
+                }
+            }
+
+            FeedResponseDiagnostics feedDiagnostics = diagnosticsAccessor
+                .getFeedResponseDiagnostics(diagnostics);
+
+            if (feedDiagnostics == null) {
+                return;
+            }
+
+            QueryInfo.QueryPlanDiagnosticsContext queryPlanDiagnostics =
+                feedDiagnostics.getQueryPlanDiagnosticsContext();
+
+            recordQueryPlanDiagnostics(queryPlanDiagnostics);
+        }
+
+        private void recordQueryPlanDiagnostics(
+            QueryInfo.QueryPlanDiagnosticsContext queryPlanDiagnostics
+        ) {
+            if (queryPlanDiagnostics == null) {
+                return;
+            }
+
+            Tags requestTags = operationTags.and(
+                createQueryPlanTags(metricTagNames)
+            );
+
+            Duration latency = queryPlanDiagnostics.getDuration();
+
+            if (latency != null) {
+                Timer requestLatencyMeter = Timer
+                    .builder(nameOf("req.gw.latency"))
+                    .description("Gateway Request latency")
+                    .maximumExpectedValue(Duration.ofSeconds(300))
```

Comment:
what would happen if the recorded value > max value? any exceptions bubble up?

Reply:
I think it just gets ignored - but let me add a unit test for it :-)